Neos.Flow/Classes/Security/Policy/PolicyService.php

```diff
@@ -314,6 +317,62 @@ public function reset(): void
         $this->roles = [];
     }
 
+    public function configurePrivilegeAnnotatedMethods(array &$policyConfiguration): void
+    {
+        $annotatedMethods = static::resolvePrivilegeAnnotatedMethods($this->objectManager);
+        foreach ($annotatedMethods as $privilegeId => $configuration) {
+            $policyConfiguration['privilegeTargets'][MethodPrivilege::class][$privilegeId]['matcher'] = $configuration['matcher'];
+
+            foreach ($configuration['roles'] as $role) {
+                $policyConfiguration['roles'][$role]['privileges'][] = [
+                    'privilegeTarget' => $privilegeId,
+                    'permission' => 'GRANT'
+                ];
+            }
+        }
+    }
+
+    /**
+     * @param ObjectManagerInterface $objectManager
+     * @Flow\CompileStatic
+     * @return array
+     */
+    protected static function resolvePrivilegeAnnotatedMethods(ObjectManagerInterface $objectManager): array
+    {
+        $privilegesWithGrantedRoles = [];
+
+        $reflectionService = $objectManager->get(ReflectionService::class);
+        $classesWithMethodsAnnotated = $reflectionService->getClassesContainingMethodsAnnotatedWith(Privilege::class);
+        foreach ($classesWithMethodsAnnotated as $className) {
+            $methodsAnnotated = $reflectionService->getMethodsAnnotatedWith($className, Privilege::class);
+            foreach ($methodsAnnotated as $methodName) {
+                $annotation = $reflectionService->getMethodAnnotation($className, $methodName, Privilege::class);
+
+                $grantedRoles = $annotation->grantedRoles;
+                $privilegeId = $annotation->id;
+
+                if ($privilegeId === null) {
+                    $packageKey = $objectManager->getPackageKeyByObjectName($className);
+                    $privilegeId = sprintf('%s:Privilege.%s', $packageKey, md5($className.'->'.$methodName));
+                }
+                $matcher = sprintf('method(%s->%s())', $className, $methodName);
+
+                if (isset($privilegesWithGrantedRoles[$privilegeId])) {
+                    if ($grantedRoles !== [] || $privilegesWithGrantedRoles[$privilegeId]['roles'] !== []) {
+                        throw new \RuntimeException(sprintf('You can not reuse a privilege id for granting roles access on a different method. Privilege id used: "%s". Method annotation: "%s->%s"', $privilegeId, $className, $methodName));
+                    }
+                    $matcher = $privilegesWithGrantedRoles[$privilegeId]['matcher'] . ' || ' . $matcher;
```

Comment:
This needs a test to verify that multiple methods with a privilegeId are all covered by the matcher.